src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs

```diff
@@ -3390,6 +3390,22 @@
+    ("SvePrefetchTest.template",         new Dictionary<string, string> { ["TestName"] = "Sve_PrefetchBytes_9",                                                                                    ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "PrefetchBytes",                                                        ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "Byte",                                                                                                                          ["LargestVectorSize"] = "64", ["NextValueOp2"] = "TestLibrary.Generator.GetByte()",   ["PrefetchOp"] = "SvePrefetchType.SV_PSTL2STRM"}),
+    ("SvePrefetchTest.template",         new Dictionary<string, string> { ["TestName"] = "Sve_PrefetchBytes_10",                                                                                   ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "PrefetchBytes",                                                        ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "Byte",                                                                                                                          ["LargestVectorSize"] = "64", ["NextValueOp2"] = "TestLibrary.Generator.GetByte()",   ["PrefetchOp"] = "SvePrefetchType.SV_PSTL3KEEP"}),
+    ("SvePrefetchTest.template",         new Dictionary<string, string> { ["TestName"] = "Sve_PrefetchBytes_11",                                                                                   ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "PrefetchBytes",                                                        ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "Byte",                                                                                                                          ["LargestVectorSize"] = "64", ["NextValueOp2"] = "TestLibrary.Generator.GetByte()",   ["PrefetchOp"] = "SvePrefetchType.SV_PSTL3STRM"}),                                                                                   
+    ("SvePrefetchTest.template",         new Dictionary<string, string> { ["TestName"] = "Sve_PrefetchInt16",                                                                                      ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "PrefetchInt16",                                                        ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "UInt16",                                                                                                                        ["LargestVectorSize"] = "64", ["NextValueOp2"] = "TestLibrary.Generator.GetUInt16()", ["PrefetchOp"] = "SvePrefetchType.SV_PLDL1KEEP"}),                                                                                  
```

Comment:
I think you can just have 4 tests for each APIs and then inside the template, create the values such taht they are valid `Sve_PrefetchBytes_` values and pass them as argument. That way we will get the random inputs.

Reply:
Which part are you referring to? The `SvePrefetchType` ?

Reply:
yes

Reply:
The problem with this is that it'll always cause the hwintrinsic to expand to the table lookup version. That's why the value is a template var to it gets expanded at the point of the hwintrinsic call.

Reply:
possibly we can make two variants of this - the one that is marked as `NoInline` and the one with `AggressiveInling`, the later will make sure that they are passed as constants.